### PR TITLE
Add Settings navigator and buttons

### PR DESCRIPTION
**Pull request by NastuzziSamy**

### CODE REVIEW

File: src/screens/Settings/SettingsScreen.js [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+	static navigationOptions = {
+		title: t('title'),
+		header: null,
+		headerForceInset: { top: 'never' },
+	};
+
+	constructor(props) {
+		super(props);
+
+		this.state = { expectedStatus: false };
```
Comment (NastuzziSamy) [inline src/screens/Settings/SettingsScreen.js]: Useless state bcs you can check de lockStatus data

File: src/screens/Settings/SettingsScreen.js [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+		);
+
+		PayUTCApi.setLockStatus(value).then(([status]) => {
+			dispatch(
+				Config.spinner({
+					visible: false,
+				})
+			);
+
+			if (status !== true && status !== false) {
```
Comment (NastuzziSamy) [inline src/screens/Settings/SettingsScreen.js]: If think you should do it in the catch method.
The API will always return the good response

File: src/screens/Settings/SettingsScreen.js [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+					{}
+				);
+				return;
+			}
+			this.onRefresh();
+		});
+	}
+
+	signOut() {
+		PayUTC.forget();
```
Comment (NastuzziSamy) [inline src/screens/Settings/SettingsScreen.js]: Use promise then => navigate

File: src/screens/Settings/SettingsScreen.js [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+				<BlockTemplate roundedTop roundedBottom shadow>
+					<Text style={{ fontSize: 22, fontWeight: 'bold', color: colors.primary }}>
+						{t('title')}
+					</Text>
+				</BlockTemplate>
+				<View style={{ height: 15 }} />
+				<SwitchBlockTemplate
+					roundedTop
+					roundedBottom
+					value={lockStatusFetching ? expectedStatus : lockStatus}
```
Comment (NastuzziSamy) [inline src/screens/Settings/SettingsScreen.js]: I think here, you should load the `lockStatus` in the AppLoader, so you are sure of the result.
+ `lockStatus` changes only when the redux promise ended

File: src/screens/Settings/SettingsScreen.js [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+					onValueChange={this.onLockChange}
+					tintColor={colors.less}
+					disabled={lockStatusFetching}
+				>
+					<View style={{ flex: 1, flexDirection: 'column' }}>
+						<Text
+							style={{
+								fontSize: 16,
+								fontWeight: 'bold',
+								color: lockStatusFetching ? colors.disabled : colors.secondary,
```
Comment (NastuzziSamy) [inline src/screens/Settings/SettingsScreen.js]: Here it is quite good used :)

File: src/screens/Settings/SettingsScreen.js (hunk 4 above)
Comment (NastuzziSamy) [reply src/screens/Settings/SettingsScreen.js]: Actually, if you disable just the button it is quite good

File: src/screens/Settings/SettingsScreen.js (hunk 2 above)
Comment (NastuzziSamy) [reply src/screens/Settings/SettingsScreen.js]: Do not forget to set the spinner no more visible if you catch an error :p

Comment (aarrtthhuurr) [reply src/screens/Settings/SettingsScreen.js]: Catch is never thrown with this API request

File: src/screens/Settings/SettingsScreen.js [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+				<BlockTemplate roundedTop roundedBottom shadow>
+					<Text style={{ fontSize: 22, fontWeight: 'bold', color: colors.primary }}>
+						{t('title')}
+					</Text>
+				</BlockTemplate>
+				<View style={{ height: 15 }} />
+				<SwitchBlockTemplate
+					roundedTop
+					roundedBottom
+					value={lockStatusFetching ? false : lockStatus}
```
Comment (NastuzziSamy) [inline src/screens/Settings/SettingsScreen.js]: Here you can use the redux default value, check my next comm :)

File: src/screens/Settings/SettingsScreen.js [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+			</ScrollView>
+		);
+	}
+}
+
+const mapStateToProps = ({ payutc }) => {
+	const lockStatus = payutc.getLockStatus();
+
+	return {
+		lockStatus: lockStatus.getData(),
```
Comment (NastuzziSamy) [inline src/screens/Settings/SettingsScreen.js]: `getData(false)` so, if the data is not yet fetched, it will return `false` by default :)

**Pull request merged**